coc/abc.py

```diff
@@ -50,48 +53,52 @@ class BaseClan:
     level: :class:`int`
         The clan's level.
     """
+    __slots__ = ("tag", "name", "_client", "badge", "level", "_response_retry", "_raw_data")
 
-    __slots__ = ("tag", "name", "_client", "badge", "level", "_response_retry")
+    def __init__(self, *, data, client, **_):
+        self._client = client
+
+        self._response_retry = data.get("_response_retry")
+        self.tag = data.get("tag")
+        self.name = data.get("name")
+        self.badge = try_enum(Badge, data=data.get("badgeUrls"),
+                              client=self._client)
+        self.level = data.get("clanLevel")
+        self._raw_data = data if client and client.raw_attribute else None
 
     def __str__(self):
         return self.name
 
     def __repr__(self):
-        return "<%s tag=%s name=%s>" % (self.__class__.__name__, self.tag, self.name)
+        return "<%s tag=%s name=%s>" % (
```

Comment:
Since we are starting to update some string formatting with f-string, we should do it everywhere. But only if we need to update the string. What do you thing? This string was updated so maybe throw in the f-string in there.